Feature/set preview limit

xtomflo opened the pull request:
Added the set preview limit, worked as far as I've tested it

Removed redundant call of  setup_logger() as it was resulting in  triplicated logging output.  ie everything that was logged after it, was printed 3 times.

sungchun12 committed:
sketches

sungchun12 committed:
save this draft

sungchun12 committed:
run on save example

sungchun12 committed:
new query handler

sungchun12 committed:
working draft

sungchun12 committed:
great working draft

sungchun12 committed:
default selection

sungchun12 committed:
dynamic metrics

sungchun12 committed:
fix duckdb storage error

sungchun12 committed:
toggle

sungchun12 committed:
toggle button

sungchun12 committed:
iteration v1

sungchun12 committed:
iteration v2

sungchun12 committed:
overlay chart working

sungchun12 committed:
fixed warnings

sungchun12 committed:
clean up legend and naming

sungchun12 committed:
reorder selections new to old

sungchun12 committed:
update naming

sungchun12 committed:
rename file

sungchun12 committed:
v1 of unified fst start

sungchun12 committed:
working unified fst start

sungchun12 committed:
better error handling

sungchun12 committed:
add new basepath name

sungchun12 committed:
trim the fat

sungchun12 committed:
color code the bars

sungchun12 committed:
color coded bars

sungchun12 committed:
save space

sungchun12 committed:
add store failures flag

sungchun12 committed:
rename folder

sungchun12 committed:
add compiled path

sungchun12 committed:
subfolders detected now

sungchun12 committed:
working query runner

sungchun12 committed:
ad hoc query runner working

sungchun12 committed:
get basic sql highlights

sungchun12 committed:
real time query runner

sungchun12 committed:
cache results

sungchun12 committed:
dynamic duckdb file path

sungchun12 committed:
better instructions

sungchun12 committed:
close the connection after fetching

sungchun12 committed:
better theme

sungchun12 committed:
remove old code

sungchun12 committed:
cell highlights v1

sungchun12 committed:
v2 of highlighting

sungchun12 committed:
works on first column now

sungchun12 committed:
improve perf

sungchun12 committed:
add another cache

sungchun12 committed:
remove numpy

sungchun12 committed:
pythonic ftw

sungchun12 committed:
consolidate functions

sungchun12 committed:
fix type check error

sungchun12 committed:
better default sql

sungchun12 committed:
more organizaed

sungchun12 committed:
add type hints

sungchun12 committed:
add more type hints

sungchun12 committed:
fix datetime attr

sungchun12 committed:
type safe and cleanup

sungchun12 committed:
unused import

sungchun12 committed:
fix metrics db error

sungchun12 committed:
fix timestamps

sungchun12 committed:
tooltips

sungchun12 committed:
more TODOs

sungchun12 committed:
radio button option

sungchun12 committed:
multi-select

sungchun12 committed:
expander feels better

sungchun12 committed:
update language

sungchun12 committed:
fun functionality

sungchun12 committed:
add diff_viewer

sungchun12 committed:
done

sungchun12 committed:
make it utc

sungchun12 committed:
v1 slider

sungchun12 committed:
remove old code

sungchun12 committed:
fix slide order index

sungchun12 committed:
timestamp function

sungchun12 committed:
kind of works

sungchun12 committed:
working version

sungchun12 committed:
done

sungchun12 committed:
query expander

sungchun12 committed:
clarify

sungchun12 committed:
fix chart with 1 iteration only

xtomflo committed:
Merge branch 'main' into feature/set-preview-limit

xtomflo committed:
Add cli command prompt and default

xtomflo committed:
adding limit var to the preview setting

xtomflo committed:
Fix the setting, cleaner implementation

xtomflo committed:
refactor(main.py): remove redundant logger instantiation and setup_logger call in start_directory_watcher function
feat(main.py): add logger instantiation in main function

sungchun12 commented:
Cool I'll take a deeper look at this on Monday! Thanks for the PR. Love that we're evolving this together :)

sungchun12 commented on fst/main.py:
```diff
@@ -22,7 +24,6 @@ def start_streamlit() -> None:
     subprocess.run(["streamlit", "run", streamlit_app_path])
 
 def start_directory_watcher(path: str, log_queue: multiprocessing.Queue) -> None:
```
```suggestion
def start_directory_watcher(path: str, log_queue: multiprocessing.Queue) -> None:
    setup_logger(log_queue)
```

This is needed or else none of the logs show up. 

before

<img width="960" alt="image" src="https://user-images.githubusercontent.com/19176976/229324732-75bcafa6-5770-4d7d-aef4-6b291df45f67.png">


after

<img width="998" alt="image" src="https://user-images.githubusercontent.com/19176976/229324737-db7c27e2-5866-4553-9547-a31626bbb219.png">

sungchun12 replied:
Show me a screenshot where you're seeing duplicate logs

sungchun12 replied:
My gut tells me this is a quirk using WSL. See if you can run this on linux using the github codespace.

xtomflo replied:
Which logs do you have in mind? 
In the screenshots you're showing different parts of the output in different scenarios, one is failure and one is success.
If it's about 'dbt build' logs (I assume it's this one), then in the second one they're also visible ("Completed successfully").

If it's about the query & build time, they're not visible in case of failure independent of whether 'setup_logger' is added here or not.

xtomflo replied:
Here with the _setup_logger()_ in _start_directory_watcher_  ( duplicated output)
![image](https://user-images.githubusercontent.com/9139386/231143360-caaf91ad-ccdd-4793-978c-094283644682.png)
![image](https://user-images.githubusercontent.com/9139386/231143216-95f18b09-1e4f-4406-b33c-c7bc7323da9a.png)

Here without - current state of the PR:
![image](https://user-images.githubusercontent.com/9139386/231143740-e5bd0d31-c578-45a1-a566-d916a95e21bf.png)

sungchun12 replied:
That's so interesting. I've never seen the duplicate logs before in my setup. This is affirming my point that something may be going on in  WSL

xtomflo replied:
double checked in Codespace and the result is similar, instead of triplicating the log output, it's duplicated now (and without the _setup_logger()_ in _start_directory_watcher_ it runs correctly),  super weird. I wonder how it's possible that on Mac you don't get the log outputs without the duplicated logging setup.  
It's being called already in main() which should be enough for the logs to show up. 
Here the screens

![image](https://user-images.githubusercontent.com/9139386/232077117-0bb2cb5b-fc5e-404c-863b-d35b3118fdb5.png)


![image](https://user-images.githubusercontent.com/9139386/232077028-e04eae16-c4c7-4a6a-a32c-1f59b395ef3e.png)

sungchun12 replied:
I'll check on codespaces once you update the code for my review. I wouldn't be surprised if there's a confounding variable in my local setup that's preventing duplicates logs.

sungchun12 commented on fst/main.py:
```diff
@@ -8,9 +8,11 @@
+from fst.config_defaults import CURRENT_WORKING_DIR, CONFIG
 
 
+logger = logging.getLogger(__name__)
```
```suggestion
```

sungchun12 commented on fst/main.py:
```diff
@@ -48,7 +49,14 @@ def listener_process(queue: multiprocessing.Queue) -> None:
+    "--preview-limit",
+    default=CONFIG["PREVIEW_LIMIT_ROWS"],
+    type=int,
+    help="Set the number of rows to preview in the UI. Defaults to 5.",
```
```suggestion
    help="Set the number of rows to preview in the terminal and copilot UI. Defaults to 5.",
```

sungchun12 commented:
The preview limit isn't working for me because `from fst.config_defaults import CONFIG` on L20 is read by the `query_handler.py` before it can be reset with the `--preview-limit` flag. I recommend following some general guidance from GPT4 below that'll point you in the right direction.

<img width="856" alt="image" src="https://user-images.githubusercontent.com/19176976/229325450-2a93be10-4723-4fe5-af69-14347d21b27b.png">
<img width="717" alt="image" src="https://user-images.githubusercontent.com/19176976/229325469-6bddea00-85a1-434e-9342-3d3cffffcfab.png">

sungchun12 reviewed (changes requested):
See my earlier feedback to refactor this and then I'll review it again!

xtomflo committed:
Update fst/main.py

Co-authored-by: Sung Won Chung <sungwonchung3@gmail.com>

xtomflo committed:
Update fst/main.py

Co-authored-by: Sung Won Chung <sungwonchung3@gmail.com>

sungchun12 commented:
I'm going to hold off on merging this until 4/19 because I want to present a couple wild features to my company on 4/18. I'll give you a shoutout for all your help!

xtomflo committed:
remove config setting

xtomflo commented:
sounds good to me! thanks a lot for the shoutout :)  I'm traveling at the moment so out of the working routine but will be back next week to make some more progress. good luck with the presentation!

xtomflo commented:
> The preview limit isn't working for me because `from fst.config_defaults import CONFIG` on L20 is read by the `query_handler.py` before it can be reset with the `--preview-limit` flag. I recommend following some general guidance from GPT4 below that'll point you in the right direction.

This is strange as well. It's working on WSL and it's working in Codespaces as well as expected.  I'm still updating to your suggestion, it's indeed a bit cleaner

sungchun12 commented on fst/config_defaults.py:
```diff
@@ -2,7 +2,9 @@
 
 CURRENT_WORKING_DIR = os.getcwd()
-
+CONFIG = {
```
You shouldn't need this anymore because the default will be 5 according to what you defined in the cli defaults

sungchun12 commented on fst/query_handler.py:
```diff
@@ -17,9 +17,14 @@
 logger = logging.getLogger(__name__)
 
+# Access the updated value of PREVIEW_LIMIT_ROWS
+preview_limit_rows = CONFIG["PREVIEW_LIMIT_ROWS"]
```
don't need this

sungchun12 commented on fst/query_handler.py:
```diff
@@ -97,7 +105,7 @@ def handle_query(query, file_path):
                         compiled_query = file.read()
                     duckdb_file_path = get_duckdb_file_path()
-                    _, column_names = execute_query(compiled_query, duckdb_file_path)
+                    _, column_names = execute_query(compiled_query, duckdb_file_path, preview_limit_rows)
```
Have the click cli default define this `preview_limit_rows` and have it pass through into the directory watcher.

xtomflo committed:
style(main.py): fix indentation and remove unused imports

refactor(query_handler.py): remove unused imports and variables, reformat code

xtomflo committed:
refactor(logger.py): prevent adding duplicate handlers to logger

xtomflo committed:
refactor(config_defaults.py): remove unused CONFIG variable

xtomflo committed:
feat(fst): add support for rows preview limit in query handler

- Add `rows_preview_limit` parameter to `start_directory_watcher` function in `main.py`
- Add `rows_preview_limit` parameter to `DynamicQueryHandler` class in `query_handler.py`
- Pass `rows_preview_limit` parameter to `handle_query` function in `query_handler.py`
- Use `rows_preview_limit` parameter instead of `preview_limit_rows` variable in `handle_query` function in `query_handler.py`

xtomflo commented:
Added a bunch of changes and should be ready for review. 
The duplicated log outputs should be resolved by a check in setup_logger() for an already existing handler. 
rows_preview_limit is passed through to the query handler without setting config variable

sungchun12 commented:
Lovely! I'll take a look this week and add comments!

sungchun12 commented:
Can you show me a screenshot of this working?

I can't get it to work at all with or without the preview flag: `fst start` `fst start --preview-limit 10`. 

Please resolve the conflicts of the files above listed by Github!

sungchun12 commented on fst/db_utils.py:
```diff
@@ -7,9 +7,11 @@
 
 @lru_cache(maxsize=128)
-def execute_query(query: str, db_file: str) -> Tuple[List[Tuple[Any]], List[str]]:
+def execute_query(query: str, db_file: str,  limit:int = 5) -> Tuple[List[Tuple[Any]], List[str]]:
```
I like this

xtomflo committed:
Merge branch 'main' into feature/set-preview-limit

xtomflo committed:
Adding fixes for M1
